### PR TITLE
Enable golangci-lint

### PR DESCRIPTION
**Pull request by metarsit**
# Changes
1. Enable all golanglint-ci linting
1. Delete commented code
1. Clean up the way we return
1. More thorough test assert

# Note
@kyoungho Please review this very carefully

### CODE REVIEW

**Comment by metarsit** [comment]
@kyoungho Do we still have travis? I have not set up my machine to run the test locally. I think we should also dockerize our testing.

File: rticonnextdds_connector.go [hunk 1]
```diff
@@ -71,39 +72,43 @@ type Infos struct {
...
-	WriterGuid     [16]byte `json:"writer_guid"`
+	WriterGUID     [16]byte `json:"writer_guid"`
 	SequenceNumber uint     `json:"sequence_number"`
 }
 
 // SampleHandler is an User defined function type that takes in pointers of
 // Samples and Infos and will handle received samples.
 type SampleHandler func(samples *Samples, infos *Infos)
 
+const (
```
Comment (kyoungho) [inline rticonnextdds_connector.go:83]: @metarsit can you add DDSRetCodeNoData = 11?

File: rticonnextdds_connector.go [hunk 2]
```diff
@@ -201,49 +201,36 @@ func (connector *Connector) Delete() (err error) {
...
 	}
-
-	retcode := int(C.RTIDDSConnector_wait(unsafe.Pointer(connector.native), (C.int)(timeoutMs)))
-	if retcode == 10 /* DDS_RETCODE_TIMEOUT */ {
-		err = errors.New("Timeout")
-		return err
-	} else if retcode != 0 /* DDS_RETCODE_OK */ {
-		err = errors.New("RTIDDSConnector_wait error")
-		return err
+	switch int(C.RTIDDSConnector_wait(unsafe.Pointer(connector.native), C.int(timeoutMs))) {
```
Comment (kyoungho) [inline rticonnextdds_connector.go:226]: @metarsit Can we also add the following?

case DDSRetCodeNoData:
  return errors.New("no data")

**Comment by kyoungho** [comment]
> @kyoungho Do we still have travis? I have not set up my machine to run the test locally. I think we should also dockerize our testing.

@metarsit Yes, we still have travis. It seems failing because some functions in use are not in the assert package. 
https://travis-ci.org/github/rticommunity/rticonnextdds-connector-go/jobs/736849408

File: rticonnextdds_connector.go (hunk 1 above)
Comment (metarsit) [reply rticonnextdds_connector.go:83]: Added

File: rticonnextdds_connector.go (hunk 2 above)
Comment (metarsit) [reply rticonnextdds_connector.go:228]: Added

**Comment by metarsit** [comment]
@kyoungho Travis takes forever to start up. 
So far, GO >1.13 are good. I have removed older GO version.

**Comment by metarsit** [comment]
@kyoungho I have updated a few more things to have it compile. We should really ditch older GO version. 

Also, the coverage drops because i reduced the number of lines of codes => I will open another PR after this writing test for those that are not covered yet :) Let's aim for >80%

**Comment by metarsit** [comment]
@kyoungho This PR's Travis has passed: https://travis-ci.org/github/rticommunity/rticonnextdds-connector-go/builds/739788616

We should be good to merge this PR if you are.

**Pull request merged**